lib/tpm2_options.h

```diff
@@ -45,8 +45,12 @@ union tpm2_option_flags {
         UINT8 verbose : 1;
         UINT8 quiet   : 1;
         UINT8 enable_errata  : 1;
+        struct {
+            char *name;
+            char *opts;
+        } tcti;
     };
-    UINT8 all;
+    UINT8 all : 3;
```

Comment:
Why did you change this?

Reply:
I moved sapi initialization to avoid a run-time dependency on libsapi. However, it doesn't matter in my use case (yet anyway), so I could go either way on this point.

I added an explicit 3-bit deceleration for `all`, since the rest of the bits in this data structure are char*'s. I agree it's a bit weird to add strings to the flags struct. I can move it if there's a better/preferred place.

Reply:
The three bit bitfield on all, kill it off. It will likely generate more code as the compiler cant just write 0 to a byte address, it would have to mask off and clear only those bits... ie read a byte, and them with 0x7 and write it back.

For the sapi split, I am not worried about the link dependency, so you'll have to change that up to the tpm2_options_new flag approach.

Otherwise, once those are done, ill be happy to pull this in.

Reply:
FYI, you can use: https://github.com/intel/tpm2-tools/pull/767 as a sample to port to master. This may or may not be done by the time you get back to this.